hostfuncs.go

```diff
@@ -0,0 +1,69 @@
+package basm
+
+import (
+	"errors"
+	"runtime"
+)
+
+//go:wasmimport env bufferLog
+func _hostFuncBufferLog(ptr, size uint32)
+
+func hostFuncBufferLog(msg string) {
+	msgData := []byte(msg)
+	inOffset, inLen := bytesToOffsetSize(msgData)
+	_hostFuncBufferLog(inOffset, inLen)
+	runtime.KeepAlive(msgData)
+}
+
+//go:wasmimport env httpRequest
+func _hostFuncHTTPRequest(offset, size uint32) uint64
+
+func hostFuncHTTPRequest(
+	input httpRequestInput,
+) (
+	httpRequestOutput,
+	error,
+) {
+	inputData, err := marshal(input)
+	if err != nil {
+		msg := "marshaling input data: " + err.Error()
```

Comment:
So that we still get the same benefits of wrapping errors without reflection (.. I think...) consider doing something like `errors.Join(err, errors.New("unmarshaling input data"))`

If it works here, there are a few other places in which we can use it and may just want to do something like:
```
func errWrap(msg string, err error) error ...
```

Reply:
updated as you suggested. It ended up being a nice place to put comment explaining why we were avoiding fmt